### PR TITLE
There is less code redundancy in the error code propagators. Also cha…

### PR DESCRIPTION
**Pull request by duckrojo**
…nge .unit behavior.

 it now has a setter function that takes care of the inheritance from the parent data and need to be set by each subclass as they all need to handle different units (StdDev, Variance, ...); the previous setter is called by the setter og parent_nddata and is not anymore handled on the fly by the unit property function.

### CODE REVIEW

File: astropy/nddata/nduncertainty.py [hunk 1]
```diff
@@ -137,14 +137,18 @@ def unit(self):
...
-            if (self._parent_nddata is None or
-                    self.parent_nddata.unit is None):
-                return None
-            else:
-                return self.parent_nddata.unit
         return self._unit
 
+    @unit.setter
+    def unit(self, value):
+        """Each subclass should interpret how it transform parent's unit into the correct Uncertainty Unit (
```
Comment (mwcraig) [inline astropy/nddata/nduncertainty.py]: Please remove parenthesis at the end of this line

File: astropy/nddata/nduncertainty.py [hunk 2]
```diff
@@ -177,6 +181,14 @@ def parent_nddata(self, value):
...
             value = weakref.ref(value)
         self._parent_nddata = value
+        #set uncertainty unit to that of the parent if it was not already set.
+        if self.unit is None:
+            if self.parent_nddata.unit is None:
+                #warning("No units specified")
+                self.unit = None
+            else:
+                #warning("Setting units from that of parent")
+                self.unit = self.parent_nddata
```
Comment (mwcraig) [inline astropy/nddata/nduncertainty.py]: Doesn't this depend on what kind of uncertainty this is?

Comment (duckrojo) [reply astropy/nddata/nduncertainty.py]: This is prevented by raising NonImplemented

File: astropy/nddata/nduncertainty.py (hunk 1 above)
Comment (duckrojo) [reply astropy/nddata/nduncertainty.py]: of course :)

**Comment by mwcraig** [comment]
Thanks!

**Pull request merged**